src/services/textChanges.ts

```diff
@@ -241,6 +249,13 @@ namespace ts.textChanges {
             return this;
         }
 
+        public replaceNodes(sourceFile: SourceFile, oldNodes: Node | Node[], newNodes: Node | Node[], options: ChangeNodeOptions & ChangeNodesOptions = {}) {
+            const startPosition = getAdjustedStartPosition(sourceFile, isArray(oldNodes) ? oldNodes[0] : oldNodes, options, Position.Start);
+            const endPosition = getAdjustedEndPosition(sourceFile, isArray(oldNodes) ? lastOrUndefined(oldNodes) : oldNodes, options);

```

Comment:
To clarify, this particular call to `lastOrUndefined` never returns `undefined`?  If that's not the case, is `getAdjustedEndPosition` going to throw?